File: 1572-matrix-diagonal-sum/1572-matrix-diagonal-sum.java
class Solution {
    public int diagonalSum(int[][] mat) {
        int result = 0;

        for(int i=0; i<mat.length; i++) {
            for(int j=0; j< mat.length; j++) {
                if((i==j) || (i+j==mat.length-1)) {
                    result += mat[i][j];
                }
            }
        }
        
        return result;
    }
}

File: 1572-matrix-diagonal-sum/1572-matrix-diagonal-sum.kt
class Solution {
    fun diagonalSum(mat: Array<IntArray>): Int {
        var result = 0
        val size = mat.size - 1
        for (i in 0..size) {
            result += mat[i][i]
            if (size - i != i) result += mat[i][size - i]
        }
        return result
    }
}